plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/controllers/CDKLaunchController.java

```diff
@@ -227,24 +230,47 @@ private void processTerminated(IServer server,IProcess process, IDebugEventSetLi
 		final ControllableServerBehavior beh = (ControllableServerBehavior)JBossServerBehaviorUtils.getControllableBehavior(server);
 		new Thread() {
 			public void run() {
-				try {
-					// sleep to allow vagrant to unlock queries. 
-					Thread.sleep(1000);
-				} catch( InterruptedException ie) {}
-				
-				// Poll the server once more 
-				IStatus stat = new VagrantPoller().getCurrentStateSynchronous(getServer());
-				if( stat.isOK()) {
-					beh.setServerStarted();
-					beh.setRunMode("run");
-				} else {
-					beh.setServerStopped();
-				}	
+				handleProcessTerminated(beh);
 			}
 		}.start();
 		DebugPlugin.getDefault().removeDebugEventListener(listener);
 	}
 	
+	
+	private void handleProcessTerminated(ControllableServerBehavior beh) {
+		try {
+			// sleep to allow vagrant to unlock queries. 
+			Thread.sleep(1000);
+		} catch( InterruptedException ie) {}
+		
+		// Poll the server once more 
+		VagrantPoller vp = new VagrantPoller();
+		IStatus stat = vp.getCurrentStateSynchronous(getServer());
+		if( stat.isOK()) {
+			beh.setServerStarted();
+			beh.setRunMode("run");
+		} else {
+			// The vm is now in a confused state.  
+			if( vp.getPollingException() instanceof OpenShiftNotReadyPollingException) {
+				// The vm is running but openshift isn't available.  
+				handleOpenShiftUnavailable(beh, (OpenShiftNotReadyPollingException)vp.getPollingException());
+			} else {
+				beh.setServerStopped();
+			}
+		}
+	}
+	
+	private void handleOpenShiftUnavailable(final IControllableServerBehavior beh, final OpenShiftNotReadyPollingException osnrpe) {
+		// Log error?  Show dialog?  
+		((ControllableServerBehavior)beh).setServerStarted();
+		new Job(osnrpe.getMessage()) {
```

Comment:
It doesn't log the message.  It pops it up in an error dialog. I did that to avoid having to make a UI extension point, or somehow mixing my core / ui code together.